slsim/Sources/supernovae.py

```diff
@@ -0,0 +1,236 @@
+import numpy as np
+import sncosmo
+from astropy import cosmology
+
+_ABSOLUTE_MAG_DISTS = {
+    "Ia": [-19.37, 0.47],
+    "Ib": [-17.90, 0.90],
+    "Ic": [-18.30, 0.60],
+    "IIb": [-17.03, 0.93],
+    "IIL": [-17.98, 0.90],
+    "IIP": [-16.80, 0.97],
+    "IIn": [-18.62, 1.48],
+    "Ic-BL": [-18.30, 0.60],
+}
+
+
+def get_accepted_sn_types():
+    all_models = sncosmo.registry._get_registry(sncosmo.Source)
+    all_models_type_dict = {
+        mod[0]: all_models._loaders[mod][2]["type"].split()[-1]
+        for mod in all_models._loaders.keys()
+    }
+    return all_models_type_dict, list(np.unique(list(all_models_type_dict.values())))
+
+
+class Supernova(sncosmo.Model):
+    """Class describing a supernova."""
+
+    def __init__(
+        self,
+        source,
+        redshift,
+        sn_type,
+        absolute_mag=None,
+        absolute_mag_band=None,
+        peak_apparent_mag=None,
+        peak_apparent_mag_band=None,
+        mag_zpsys="AB",
+        cosmo=cosmology.FlatLambdaCDM(H0=70, Om0=0.3),
+        **kwargs
+    ):
+        """
+
+        :param source: The model for the spectral evolution of the source. If a string
+            is given, it is used to retrieve a `~sncosmo.Source` from
+            the registry.
+        :type source: `~sncosmo.Source` or str
+        :param redshift: The redshift of the source.
+        :type redshift: float
+        :param sn_type: Supernova type (Ia, Ib, Ic, IIP, etc.)
+        :type sn_type: str
+        :param absolute_mag: Absolute magnitude of the supernova
+        :type absolute_mag: float
+        :param absolute_mag_band: Band used to normalize to absolute magnitude
+        :type absolute_mag_band: str or `~sncosmo.Bandpass`
+        :param peak_apparent_mag: Peak apparent mag of the supernova
+        :type peak_apparent_mag: str or `~sncosmo.Bandpass`
+        :param peak_apparent_mag_band: Band used to normalize to apparent magnitude
```

Comment:
Can we pass LSST filter to this class so that we get apparent magnitude in LSST fiters?